google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java

```diff
@@ -378,6 +380,40 @@ public long writeFileToTable(String datasetName, String tableName, Path csvPath)
     // [END writeFileToTable]
   }
 
+  /**
+   * Example of writing a newline-delimited-json file with textual fields from GCS to a table.
+   */
+  // [TARGET writer(WriteChannelConfiguration)]
+  // [VARIABLE "my_dataset_name"]
+  // [VARIABLE "my_table_name"]
+  // [VARIABLE "my_source_uri")]
+  // [VARIABLE "field_names")]
+  public Long writeRemoteFileToTable(String datasetName, String tableName, String sourceUri, List<String> fieldNames)
+      throws InterruptedException {
+    // [START bigquery_load_table_gcs_json]
+    TableId tableId = TableId.of(datasetName, tableName);
+    LoadJobConfiguration configuration = LoadJobConfiguration.builder(tableId, sourceUri)
+        .setFormatOptions(FormatOptions.json())
+        .build();
+    // Table field definition
+    ArrayList<Field> fields = new ArrayList<>();
+    for (String fieldName: fieldNames) {
+      fields.add(Field.of(fieldName, LegacySQLTypeName.STRING));
+    }
+    // Table schema definition
+    Schema schema = Schema.of(fields.toArray(new Field[fields.size()]));
+    // Create the table
+    StandardTableDefinition tableDefinition = StandardTableDefinition.of(schema);
+    bigquery.create(TableInfo.of(tableId, tableDefinition));
```

Comment:
You shouldn't have to create the table manually. The load job should create the table for you.

Reply:
I am looking into it to see how I can modify the code to let the load job create the table.